src/PrestaShopBundle/Resources/views/Admin/Component/Layout/header.html.twig

```diff
@@ -37,14 +37,14 @@
     var customer_name_msg = '{{ 'Customer name:'|trans({}, 'Admin.Navigation.Header') }} ';
     var new_msg = '{{ 'A new message was posted on your store.'|trans({}, 'Admin.Navigation.Header') }}';
     var see_msg = '{{ 'Read this message'|trans({}, 'Admin.Navigation.Header') }}';
-    var token = '{{ token|e('js') }}';
-    var currentIndex = '{{ currentIndex|e('js') }}';
+    var token = '{{ this.token|e('js') }}';
+    var currentIndex = '{{ this.currentIndex|e('js') }}';
     var employee_token = '{{ this.employeeToken|e('js') }}';
     var choose_language_translate = '{{ 'Choose language:'|trans({}, 'Admin.Actions') }}';
-    var default_language = '{{ default_language|intCast }}';
-    var admin_modules_link = '{{ link.getAdminLink("AdminModulesSf", true, {'route': "admin_module_manage"})|e('js')}}';
-    var admin_notification_get_link = '{{ link.getAdminLink("AdminCommon")|e('js') }}';
-    var admin_notification_push_link = adminNotificationPushLink = '{{ link.getAdminLink("AdminCommon", true, {'route': 'admin_common_notifications_ack'})|e('js') }}';
+    var default_language = '{{ this.defaultLanguage|intCast }}';
+    var admin_modules_link = '{{ this.link.getAdminLink("AdminModulesSf", true, {'route': "admin_module_manage"})|e('js')}}';
+    var admin_notification_get_link = '{{ this.link.getAdminLink("AdminCommon")|e('js') }}';
```

Comment:
You're not supposed to inject the Link component We're supposed to get as far away as possible from the legacy code Even when we generate legacy URLs

@boherm created a twig function to replace this https://github.com/PrestaShop/PrestaShop/blob/c47d10abffb64c4ea9cf479d25ae0f88cad2853b/src/PrestaShopBundle/Twig/Extension/PathExtension.php#L53 But it should be refactored and renamed

We can't use the same function as the original twig one or we simply can't tell them apart which is gonna be a mess for future refacto and replacement

Reply:
So what you need to do here is:
- rework the added twig function `path` into a dedicated `legacy_path`
- it should handle the generation of legacy links ONLY
- you'll need to replace the usages of `path` in all the other component templates that already depend on them
- finally you can use this function in your template here to correctly generate a legacy URL (and remove the link injection in the template)